### PR TITLE
Add batch actions for accounts in admin UI

### LINKED ISSUE #11346: Batch account management
### Pitch

It would be great to have the ability to perform sorting and batch operations on multiple accounts at the same time in the admin Accounts interface. (There are many available examples; most software that allows public user registration includes this kind of tool in administration and Mastodon already has this kind of interface for users to manage their following :) )

### Motivation

Allowing accounts created by automated registration to remain both artificially inflates user numbers and creates a potential security risk since these accounts are active and could be leveraged at any time. 

Right now account moderation requires manually scrolling through every page of the Accounts list (there is no sorting by e.g. activity or permissions available to reduce or reorder the list) and opening each individual account page and performing an administration function. This creates an unreasonable administration cleanup burden for e.g. situations with a high volume of bot spam registrations. (Ain't no one got time for that ;) )

Example: we still have multiple pages of this in our active accounts because there are no available cleanup tools for this kind of attack: 
spamreg

### LINKED ISSUE #16124: Record account approval/rejection in the moderator audit log
### Pitch

If an instance has the **Approval required to sign up** option set in the admin interface, all new users are obviously required to be approved or rejected when they sign up. Problem is, when an account is approved or rejected by a mod, this is **NOT** recorded in the moderator audit log, so admins have no viability of who approved or rejected an account.

### Motivation

Sometimes accounts can be approved or rejected incorrectly. This will allow for admins to look back and coach a mod on their rationale for approval or rejection.

All activity a user with elevated privileges should be logged; account approval/rejection is a very important part of that.

### PR DESCRIPTION
**Pull request by Gargron**
- Fix #11346
- Fix #16124

New look for admin accounts listing. This emphasizes information that may help weed out "real" accounts from "spam" accounts when doing batch actions (avatar, display name, followers count, posts count), and adds the e-mail domain as well as the IP (both can be clicked to filter accounts by that e-mail domain/IP). Color coding that's consistent with designs in the trends area help see pending accounts and suspended accounts.

image

Previously separate "pending accounts" area has been folded into the main account listing. This means "approve all" and "reject all" buttons are no longer available, but I think they were not massively more useful than batch actions themselves.

Approve and reject actions are now logged into the audit log.

IPs in the IP rules listing now link to the accounts listing filtered by that IP.

### CODE REVIEW

**Review by ykzts — APPROVED** [review]

File: app/helpers/admin/dashboard_helper.rb [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+      '-'
+    end
+  end
+
+  def relevant_account_timestamp(account)
+    timestamp, exact = begin
+      if account.user_current_sign_in_at && account.user_current_sign_in_at < 24.hours.ago
+        [account.user_current_sign_in_at, true]
+      elsif account.user_current_sign_in_at
+        [account.user_current_sign_in_at, false]
```
Comment (ClearlyClaire) [inline app/helpers/admin/dashboard_helper.rb:26]: Might be slightly confusing if it shows “Today” for a log in that was actually 23h ago. Not sure how it's handled elsewhere.

File: app/models/form/account_batch.rb [hunk 2]
```diff
@@ -25,19 +26,21 @@ def save
       suppress_follow_recommendation!
     when 'unsuppress_follow_recommendation'
       unsuppress_follow_recommendation!
+    when 'suspend'
+      suspend!
```
Comment (ClearlyClaire) [inline app/models/form/account_batch.rb:30]: I know this is not a new thing but I'm slightly uncomfortable with having moderation actions and individual follower/following management in the same class.

File: app/models/form/account_batch.rb [hunk 3]
```diff
@@ -25,19 +26,21 @@ def save
...
     end
   end
 
   private
 
   def follow!
-    accounts.find_each do |target_account|
+    accounts.each do |target_account|
```
Comment (ClearlyClaire) [inline app/models/form/account_batch.rb:37]: Why this change? Especially since `approve!` is changed from a `each` to a `find_each`.

File: app/models/form/account_batch.rb [hunk 4]
```diff
@@ -61,23 +64,31 @@ def accounts
   end
 
   def approve!
-    users = accounts.includes(:user).map(&:user)
-
-    users.each { |user| authorize(user, :approve?) }
-         .each(&:approve!)
+    accounts.includes(:user).find_each do |account|
+      approve_account(account)
+    end
```
Comment (ClearlyClaire) [inline app/models/form/account_batch.rb:69]: I think the rationale behind having the `authorize` calls first was to make sure we don't have a partial application of the batch actions. In practice, I don't think we can have the `authorize` call work for a user and fail for another, though.

File: app/models/form/account_batch.rb [hunk 5]
```diff
@@ -61,23 +64,31 @@ def accounts
...
-    records.each { |account| authorize(account.user, :reject?) }
-           .each { |account| DeleteAccountService.new.call(account, reserve_email: false, reserve_username: false) }
+  def suspend!
+    accounts.find_each do |account|
+      if account.user_pending?
+        reject_account(account)
+      else
+        suspend_account(account)
+      end
+    end
```
Comment (ClearlyClaire) [inline app/models/form/account_batch.rb:85]: Partial application of a `suspend!` action seem possible, if one of the batched accounts is an admin.

**Review by ClearlyClaire — COMMENTED** [review]
Overall an improvement, have a few questions and remarks though.

As for the color coding, it can be useful, but I don't think it's explicit enough: there is nothing in the UI that explicitly tells these accounts are pending, admins have to guess that's what the color means.

File: app/helpers/admin/dashboard_helper.rb (hunk 1 above)
Comment (Gargron) [reply app/helpers/admin/dashboard_helper.rb:26]: The reason I did it this way is we only bump current_sign_in_at once every 24 hours. So if it's less than 24h ago, it may give a misleading sense of precision.

File: app/models/form/account_batch.rb (hunk 3 above)
Comment (Gargron) [reply app/models/form/account_batch.rb:37]: Realistically there is no difference, the account_ids array will never be > 1000 accounts for this to make a difference, but CodeClimate complains about using each. I don't know why it didn't complain here. I originally changed all find_each to each, and this apparently got left behind.

File: app/models/form/account_batch.rb (hunk 4 above)
Comment (ClearlyClaire) [reply app/models/form/account_batch.rb:69]: Yeah that's what I said below, but here it's for approve/reject
